sys/events.js

```diff
@@ -85,6 +85,26 @@ EventService.prototype._emit = function(hyper, req) {
                 return event;
             })
             .filter(function(event) { return !!event; });
+
+            // Change-propagation will set up the x-rerender-reason header, indicating
+            // the event which caused the rerender. In case RESTBase is about to emit
+            // the same event, it will cause a rerender loop. So, log an error and skip
+            // an event.
+            var reRenderReason = req.headers && req.headers['x-rerender-reason']
```

Comment:
Minor nit: `reRenderReason` looks ... ugh, unnatural to me. `origReason` perhaps?